packages/shared/lib/services/sync/config/deploy.service.ts

```diff
@@ -834,7 +826,7 @@ async function compileDeployInfo({
         ...flow,
         name: syncName,
         version
-    });
+    } as SyncDeploymentResult); // TODO: remove this as
```

Comment:
you are casting type and sync_type above why not here as well. 

Anyway I think the weirdness comes from this line: https://github.com/NangoHQ/nango/blob/master/packages/shared/lib/services/sync/config/deploy.service.ts#L646 
I am not sure why the assignation is there